### PR TITLE
Move list sending

### LINKED ISSUE #7: Get client to send move list to server when "end turn" is clicked on

### LINKED ISSUE #8: Get server to accept move list from client
Acceptor function must validate list of moves and save valid moves to the global move list
We probably want something to check if everyone has submitted a move yet, and if so, have this thread resolve the move list

### PR DESCRIPTION
**Pull request by JustinHuPrime**
Implemented parsing, sending, and receiving of move lists, and added tests for serialize and parse

Closes #7 
Closes #8

### CODE REVIEW

File: src/common/Board.hs [hunk 1]
```diff
@@ -41,4 +44,85 @@ data Entity
     ---  - Drive health
     ---  - Structure health
     Ship Int Vec2 Vec2 Int String Int Bool Double Double Int Int Int
-  deriving (Show)
\ No newline at end of file
+  deriving (Show, Eq)
```
Comment (wyndro) [inline src/common/Board.hs]: Same from before? This is going to be a bit of a pain to merge huh?

File: src/common/NetInterface.hs [hunk 2]
```diff
@@ -18,11 +18,18 @@ with N-Planetary. If not, see <https://www.gnu.org/licenses/>.
...
 port = "3000"
 
--- on connection, server waits for client to say hello (expects some particular handshake string)
--- server replies with current game state
--- server waits until client sends a move
--- if everyone has sent in a move, this thread handles the list of moves
--- otherwise, this thread adds the move to the list of moves
\ No newline at end of file
+readPacket :: Socket -> IO B.ByteString
+readPacket s = do
```
Comment (wyndro) [inline src/common/NetInterface.hs]: I brought up my concern about this method in a different review.

File: src/common/Serializing.hs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+N-Planetary is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along
+with N-Planetary. If not, see <https://www.gnu.org/licenses/>.
+-}
+
+module Serializing where
```
Comment (wyndro) [inline src/common/Serializing.hs:19]: Yay! It gets its own file

File: src/common/NetInterface.hs (hunk 2 above)
Comment (wyndro) [reply src/common/NetInterface.hs]: Other than this though, it's good.

File: src/common/Board.hs (hunk 1 above)
Comment (JustinHuPrime) [reply src/common/Board.hs]: Sequencing, Bronwyn - this depends on the board state sending PR being merged first.

**Review by JohannCooper — APPROVED** [review]

**Review by wyndro — APPROVED** [review]

**Pull request merged**